src/core/MOM_open_boundary.F90

```diff
@@ -762,9 +787,22 @@ subroutine initialize_segment_data(G, GV, US, OBC, PF)
     IsdB = segment%HI%IsdB ; IedB = segment%HI%IedB
     JsdB = segment%HI%JsdB ; JedB = segment%HI%JedB
 
-    do m=1,num_fields
-      call parse_segment_data_str(trim(segstr), m, trim(fields(m)), &
-          value, filename, fieldname)
+    obgc_segments_props_list => obgc_segments_props !Get a pointer to the saved type
```

Comment:
This block of code uses module variables and it ignores the guidance in the MOM6 style guide. Please correct the indenting and white space, and avoid using saves types.